pkg/apis/duck/v1/identity_lifecycle.go

```diff
@@ -33,3 +34,13 @@ func (s *IdentityStatus) MarkWorkloadIdentityFailed(cs *apis.ConditionSet, reaso
 	// It will be counted for conditionReady only if it is failed.
 	cs.Manage(s).MarkFalse(apis.ConditionReady, "WorkloadIdentityFailed", messageFormat, messageA...)
 }
+
+func (s *IdentityStatus) MarkWorkloadIdentityUnknown(cs *apis.ConditionSet, reason, messageFormat string, messageA ...interface{}) {
+	cs.Manage(s).MarkUnknown(IdentityConfigured, reason, messageFormat, messageA...)
+	// ConditionType IdentityConfigured is not included in apis.NewLivingConditionSet{}, so it is not counted for conditionReady.
+	// Set ConditionReady to be Unknown if the initial status of ConditionReady is Ready.
+	// If the initial status of ConditionReady is not Ready, we keep it as it.
+	if c := cs.Manage(s).GetCondition(apis.ConditionReady); c != nil && c.Status == corev1.ConditionTrue {
+		cs.Manage(s).MarkUnknown(apis.ConditionReady, "WorkloadIdentityUnknown", messageFormat, messageA...)
```

Comment:
@nachocano @grac3gao There is an ongoing discussion https://github.com/google/knative-gcp/pull/1538#discussion_r465322151 about whether we should directly mark ConditionReady to be false or only mark it as Unknown when the initial condition of ConditionReady is Ready... 
Could you please check to see if this is the logic that you want, please let me know if this should be changed.

Reply:
Just synced with @grac3gao, we will do something slightly different. But basically what she said in her comment. 
Please sync with her and she can point you in the right direction